front/src/modules/simulationResult/components/SpaceTimeChart/useLazyProjectTrains.ts

```diff
@@ -1,16 +1,18 @@
 import { useEffect, useState, useCallback, useRef } from 'react';
 
-import TrainProjectionLazyLoader from 'applications/operationalStudies/helpers/TrainProjectionLazyLoader';
+import TrainProjectionLazyLoader, {
+  type ProjectionResult,
+} from 'applications/operationalStudies/helpers/TrainProjectionLazyLoader';
 import upsertNewProjectedTrains from 'applications/operationalStudies/helpers/upsertNewProjectedTrains';
-import type { ProjectPathTrainResult, ProjectPathForm } from 'common/api/osrdEditoastApi';
+import type { OccupancyBlockForm } from 'common/api/osrdEditoastApi';
 import type { TrainSpaceTimeData } from 'modules/simulationResult/types';
 import type { TimetableItemId, TimetableItem } from 'reducers/osrdconf/types';
 import { useAppDispatch } from 'store';
 
 type UseLazyProjectTrainsOptions = {
   infraId: number;
   electricalProfileSetId: number | undefined;
-  path: ProjectPathForm['path'] | undefined;
+  path: OccupancyBlockForm['path'] | undefined;
```

Comment:
can suggest because of renaming but  what about using nullable operator 

```
electricalProfileSetId?: number;
  path?: OccupancyBlockForm['path'];
```